src/blob/index/simple.rs

```diff
@@ -160,6 +186,25 @@ impl Simple {
         self.filter = Bloom::from_raw(buf_ref)?;
         Ok(())
     }
+
+    async fn read_header_from_file(file: File) -> Result<Vec<u8>> {
+        let mut buf = file.read_all().await?;
+        let end_magic: Vec<_> = buf.drain(buf.len() - std::mem::size_of::<u64>()..).collect();
```

Comment:
I think it's better to use split. And don't create new `Vec<_>` but just compare slices below